packages/js/src/containers/TwitterEditor.js

```diff
@@ -21,8 +23,22 @@ const socialMediumName = "Twitter";
  * @returns {void}
  */
 const imageCallback = ( image ) => {
+	const twitterImageType = get( window, "wpseoScriptData.metabox.twitterCardType" );
+
+	const isLarge = twitterImageType !== "summary";
+	const imageMode = isLarge ? "landscape" : "square";
+
+	const idealWidth = TWITTER_IMAGE_SIZES[ imageMode + "Width" ];
+	const idealHeight = TWITTER_IMAGE_SIZES[ imageMode + "Height" ];
+
+	const idealImageSize = Object.values( image.sizes ).find( size => {
+		return size.width >= idealWidth && size.height >= idealHeight;
+	} );
+
+	const imageUrl = idealImageSize ? idealImageSize.url : image.sizes.full.url;
```

Comment:
Seems unsafe to presume the `full` size is always present? Or is that a WP thing?

Reply:
It's a WordPress thing 🙂